pkg/sources/registration.go

```diff
@@ -0,0 +1,195 @@
+package sources
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/kubeshop/botkube/pkg/config"
+	"github.com/kubeshop/botkube/pkg/utils"
+)
+
+type Registration struct {
+	informer        cache.SharedIndexInformer
+	log             logrus.FieldLogger
+	mapper          meta.RESTMapper
+	dynamicCli      dynamic.Interface
+	events          []config.EventType
+	mappedResources []string
+	mappedEvent     config.EventType
+}
+
+func (i Registration) handleEvent(ctx context.Context, resource string, target config.EventType, sourceRoutes []Route, fn eventHandler) {
```

Comment:
Any reason for not using pointer receiver? Botkube is concurrent-heavy application, if this or parent function is called concurrently, concurrent changes will not be visible to original one. If you are sure about, you always read from receiver,  and never write to that good to go with non-pointer  receiver. Btw, if you are not sure about what kind of operations this receiver does, just use pointer receiver :) You can see Golang code review comments here https://github.com/golang/go/wiki/CodeReviewComments#receiver-type

Reply:
Yup, for now this is a pure data object. Not wanting to be heavy handed.

Happy to change if the context changes in the future.